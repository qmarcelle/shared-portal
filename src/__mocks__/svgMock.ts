import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
