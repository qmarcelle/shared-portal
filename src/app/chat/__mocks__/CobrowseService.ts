export class CobrowseService {
  constructor() {
    // Empty constructor
  }

  initialize = jest.fn().mockResolvedValue({});
  createSession = jest.fn().mockResolvedValue('test-code');
  endSession = jest.fn().mockResolvedValue({});
}
