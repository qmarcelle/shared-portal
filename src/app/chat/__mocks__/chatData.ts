import { ChatConfig, ClientType, Plan, UserEligibility } from '../models/chat';

/**
 * Mock chat configuration for testing
 */
export const mockChatConfig: ChatConfig = {
  token: 'mock-jwt-token',
  endPoint: 'https://api.example.com/chat',
  demoEndPoint: 'https://demo-api.example.com/chat',
  opsPhone: '800-123-4567',
  memberFirstname: 'John',
  memberLastname: 'Doe',
  memberId: 'MEMBER123',
  groupId: 'GROUP456',
  planId: 'PLAN789',
  planName: 'Premium Health Plan',
  businessHours: {
    isOpen24x7: true,
    days: [
      { day: 'Monday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Tuesday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Wednesday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Thursday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Friday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Saturday', openTime: '09:00', closeTime: '17:00' },
      { day: 'Sunday', openTime: '09:00', closeTime: '17:00' },
    ],
  },
  cobrowseSource: 'https://cobrowse.example.com',
  cobrowseURL: 'https://js.cobrowse.io/CobrowseIO.js',
  coBrowseLicence: 'mock-cobrowse-license-key',
};

/**
 * Mock user eligibility data for testing
 */
export const mockUserEligibility: UserEligibility = {
  isChatEligibleMember: true,
  isDemoMember: false,
  isAmplifyMem: false,
  groupId: 'GROUP456',
  memberClientID: ClientType.Default,
  getGroupType: 'Commercial',
  isBlueEliteGroup: false,
  isMedical: true,
  isDental: true,
  isVision: true,
  isWellnessOnly: false,
  isCobraEligible: false,
  chatHours: 'Monday-Friday: 8:00 AM - 6:00 PM',
  rawChatHours: 'M_F_8_6',
  isChatbotEligible: true,
  memberMedicalPlanID: 'PLAN789',
  isIDCardEligible: true,
  memberDOB: '1980-01-01',
  subscriberID: 'SUB123',
  sfx: '01',
  memberFirstname: 'John',
  memberLastName: 'Doe',
  userID: 'USER456',
  isChatAvailable: true,
  routingchatbotEligible: true,
  idCardChatBotName: 'speechstorm-chatbot',
  // Added fields for enhanced routing
  RoutingChatbotInteractionId: '',
  coverage_eligibility: '',
  lob_group: 'Commercial',
  Origin: 'member-portal',
  Source: 'web',
};

/**
 * Mock BlueCare user eligibility for testing specialized routing
 */
export const mockBlueCareUserEligibility: UserEligibility = {
  ...mockUserEligibility,
  memberClientID: ClientType.BlueCare,
  lob_group: ClientType.BlueCare,
  getGroupType: 'BlueCare',
};

/**
 * Mock Dental-only user eligibility for testing specialized routing
 */
export const mockDentalOnlyUserEligibility: UserEligibility = {
  ...mockUserEligibility,
  isMedical: false,
  isDental: true,
  isVision: false,
  isWellnessOnly: false,
  coverage_eligibility: 'dental_only',
};

/**
 * Mock ID Card request user eligibility for testing specialized routing
 */
export const mockIdCardUserEligibility: UserEligibility = {
  ...mockUserEligibility,
  isIDCardEligible: true,
  RoutingChatbotInteractionId: 'ID_CARD_REQ',
};

/**
 * Mock current plan for testing
 */
export const mockCurrentPlan: Plan = {
  planId: 'PLAN789',
  planName: 'Premium Health Plan',
  isEligibleForChat: true,
  isCurrentPlan: true,
};

/**
 * Mock available plans for testing multi-plan scenarios
 */
export const mockAvailablePlans: Plan[] = [
  {
    planId: 'PLAN789',
    planName: 'Premium Health Plan',
    isEligibleForChat: true,
    isCurrentPlan: true,
  },
  {
    planId: 'PLAN456',
    planName: 'Basic Dental Plan',
    isEligibleForChat: true,
    isCurrentPlan: false,
  },
];
