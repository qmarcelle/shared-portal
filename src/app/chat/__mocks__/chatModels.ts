// Mock models for chat tests

// Client type enum
export enum ClientType {
  BlueCare = 'BC',
  BlueCarePlus = 'DS',
  CoverTN = 'CT',
  CoverKids = 'CK',
  SeniorCare = 'BA',
  Individual = 'INDV',
  BlueElite = 'INDVMX',
  Default = 'Default',
}

// Chat type enum
export enum ChatType {
  BlueCareChat = 'BlueCare_Chat',
  SeniorCareChat = 'SCD_Chat',
  DefaultChat = 'MBAChat',
}

// Basic chat message type
export interface ChatMessage {
  id?: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp?: number;
}

// Chat session information
export interface ChatSession {
  id: string;
  active: boolean;
  agentName?: string;
  messages: ChatMessage[];
}

// Plan information
export interface PlanInfo {
  planId: string;
  planName: string;
  lineOfBusiness: string; // Maps to ClientType
  isEligibleForChat: boolean;
  businessHours?: string;
}
