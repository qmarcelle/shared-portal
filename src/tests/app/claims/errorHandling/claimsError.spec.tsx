import MyClaimsPage from '@/app/claims/page';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { createAxiosErrorForTest } from '@/tests/test_utils';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

jest.mock('src/auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        currUsr: {
          plan: { memCk: '123456789', grpId: '87898', sbsbCk: '654567656' },
        },
      },
    }),
  ),
}));

jest
  .useFakeTimers({
    doNotFake: ['nextTick', 'setImmediate'],
  })
  .setSystemTime(new Date('2024-11-28T00:00:00.000'));

describe('Claims SnapshotList Error Handling', () => {
  it('should show the error message when api fails', async () => {
    mockedAxios.get
      // LoggedIn User Info
      .mockResolvedValueOnce({
        data: {
          isActive: true,
          subscriberLoggedIn: true,
          lob: 'REGL',
          groupData: {
            groupID: '100000',
            groupCK: '21908',
            groupName: 'Chris B Hall Enterprises',
            parentGroupID: '100001',
            subGroupID: '0001',
            subGroupCK: 28951,
            subGroupName: 'Chris B Hall Enterprises',
            clientID: 'EI',
            policyType: 'INT',
            groupEIN: '620427913',
          },
          networkPrefix: 'QMI',
          subscriberID: '902218823',
          subscriberCK: '91722400',
          subscriberFirstName: 'CHRIS',
          subscriberLastName: 'HALL',
          subscriberTitle: '',
          subscriberDateOfBirth: '08/06/1959',
          subscriberOriginalEffectiveDate: '01/01/2001',
          members: [
            {
              isActive: true,
              memberOrigEffDt: '06/29/2009',
              memberCk: 91722407,
              firstName: 'CHRISTMAS',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'S',
              birthDate: '06/29/2009',
              gender: 'M',
              memberSuffix: 6,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: true,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '06/15/2009',
              memberCk: 91722406,
              firstName: 'KRISSY',
              middleInitial: 'C',
              lastName: 'HALL',
              title: '',
              memRelation: 'D',
              birthDate: '06/15/2009',
              gender: 'F',
              memberSuffix: 5,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: false,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '05/01/2009',
              memberCk: 91722405,
              firstName: 'CHRISTIAN',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'S',
              birthDate: '10/31/2011',
              gender: 'M',
              memberSuffix: 4,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: true,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '01/01/2001',
              memberCk: 91722402,
              firstName: 'KRISTY',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'W',
              birthDate: '05/17/1971',
              gender: 'F',
              memberSuffix: 1,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: true,
              esipharmacyEligible: false,
            },
            {
              isActive: true,
              memberOrigEffDt: '01/01/2001',
              memberCk: 91722401,
              firstName: 'CHRIS',
              middleInitial: 'B',
              lastName: 'HALL',
              title: '',
              memRelation: 'M',
              birthDate: '08/06/1959',
              gender: 'M',
              memberSuffix: 0,
              mailAddressType: 'H',
              workPhone: '1234567890',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'D',
                  coverageLevel: '*',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'D',
                  planID: 'DEHCNY02',
                  effectiveDate: 1509508800000,
                  planStartDate: 1451624400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'DentalBlue Preferred Network',
                },
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: true,
              hasSocial: true,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '01/01/2014',
              memberCk: 91722409,
              firstName: 'CHRISTOFF',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'S',
              birthDate: '01/01/2000',
              gender: 'M',
              memberSuffix: 8,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: false,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '12/11/2006',
              memberCk: 91722408,
              firstName: 'CHRISTO',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'S',
              birthDate: '12/11/2006',
              gender: 'M',
              memberSuffix: 7,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: true,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '12/11/2006',
              memberCk: 54363201,
              firstName: 'ChrisBalance',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'S',
              birthDate: '12/11/2006',
              gender: 'M',
              memberSuffix: 7,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: true,
              esipharmacyEligible: true,
            },
            {
              isActive: true,
              memberOrigEffDt: '12/11/2006',
              memberCk: 877944401,
              firstName: 'CHRISTOCLAIM',
              middleInitial: '',
              lastName: 'HALL',
              title: '',
              memRelation: 'S',
              birthDate: '12/11/2006',
              gender: 'M',
              memberSuffix: 7,
              mailAddressType: 'H',
              workPhone: '',
              otherInsurance: [],
              coverageTypes: [
                {
                  productType: 'M',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'S',
                  coverageLevel: 'A',
                  exchange: false,
                  indvGroupInd: '',
                  pedAdultInd: '',
                },
                {
                  productType: 'V',
                  coverageLevel: 'A',
                  exchange: true,
                  indvGroupInd: 'Group',
                  pedAdultInd: 'Adult',
                },
              ],
              planDetails: [
                {
                  productCategory: 'M',
                  planID: 'MBPX0806',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Blue Network P',
                },
                {
                  productCategory: 'S',
                  planID: 'WSXM0218',
                  effectiveDate: 1546318800000,
                  planStartDate: 1514782800000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Wellness Plan',
                },
                {
                  productCategory: 'V',
                  planID: 'VEMGN002',
                  effectiveDate: 1546318800000,
                  planStartDate: 1383278400000,
                  planClassID: 'PPOA',
                  networkPlanName: 'Vision Blue',
                },
              ],
              inXPermissions: true,
              futureEffective: false,
              loggedIn: false,
              hasSocial: true,
              esipharmacyEligible: true,
            },
          ],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
            {
              productType: 'D',
              coverageLevel: '*',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          addresses: [
            {
              type: '1',
              address1: 'TEST BLUE ACCESS',
              address2: '',
              address3: '',
              city: 'Chattanooga',
              state: 'TN',
              zipcode: '37412',
              county: 'HAMILTON',
              phone: '',
              email: '',
            },
            {
              type: '2',
              address1: 'TEST 2 BLUE ACCESS',
              address2: '',
              address3: '',
              city: 'Chattanooga',
              state: 'TN',
              zipcode: '37402',
              county: 'HAMILTON',
              phone: '',
              email: '',
            },
            {
              type: 'H',
              address1: '1 CAMERON HILL CIRCLE',
              address2: '',
              address3: '',
              city: 'CHATTANOOGA',
              state: 'TN',
              zipcode: '37402',
              county: 'HAMILTON',
              phone: '4235353065',
              email: '',
            },
          ],
          healthCareAccounts: [],
          esigroupNum: '100000MBPX0806',
          cmcondition: [],
        },
      })
      // Claims Data
      .mockRejectedValueOnce(
        createAxiosErrorForTest({ status: 400, errorObject: {} }),
      );

    // Render the page
    const Result = await MyClaimsPage({
      searchParams: new Promise((resolve) => {
        resolve({});
      }),
    });
    const { container } = render(Result);

    expect(screen.queryByText('13 Claims')).toBeNull();
    // Members Api was called
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/byMemberCk/123456789',
    );
    // Claims List APi was called
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/byMemberCk/123456789/claims?from=11-28-2022&to=11-28-2024&type=MDV&includeDependents=true',
    );

    // Error Message is shown
    expect(
      screen.getByText(
        'There was a problem loading your information. Please try refreshing the page or returning to this page later.',
      ),
    ).toBeVisible();

    expect(container).toMatchSnapshot();
  });
});
