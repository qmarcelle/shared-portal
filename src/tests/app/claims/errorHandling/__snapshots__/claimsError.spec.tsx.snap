// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claims SnapshotList Error Handling should show the error message when api fails 1`] = `
<div>
  <main
    class="flex flex-col justify-center items-center page"
  >
    <div
      class="flex flex-col app-content app-base-font-color"
    >
      <h2
        class="title-2 mb-0 !font-light !text-[32px]/[40px]"
      >
        Claims
      </h2>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
      />
      <section
        class="flex justify-start self-start"
      >
        <div
          class="flex flex-col"
        >
          <span
            class="rich-text body-1 mb-4"
          >
            <span>
              If you need more than two years of claims,
               
            </span>
            <span
              class="link font-bold"
            >
              start a chat
            </span>
            <span>
               or call us at [1-800-000-000].
            </span>
          </span>
          <span
            class="rich-text body-1 undefined"
          >
            <span>
              Need to submit a claim? 
            </span>
            <span
              class="link font-bold"
            >
              Get the form you need
            </span>
            <span>
              .
            </span>
          </span>
        </div>
      </section>
      <div
        class="card-main neutral container mt-4"
      >
        <div
          class="flex flex-row flex flex-row align-top m-4"
        >
          <img
            alt="alert"
            class="icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span
            style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
          />
          <div
            class="flex flex-row flex flex-col flex-grow"
          >
            <p
              class="body-1"
            >
              There was a problem loading your information. Please try refreshing the page or returning to this page later.
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
